### ISSUE TITLE
Ability to remove role='searchbox' and add type='search'

### ISSUE THREAD

**Issue by eivindland** [microsoft/fluentui#14331]
### Describe the issue:

In accessbility testing the searchbox-component did not perform well on various screen reader software/web browsers. (Different from Egde/Narrator). Our tester suggested removing role='searchbox' and add type='search' instead for better overall screen reader and browser support. 

Is it possible to add a prop to override/remove the default role='searchbox' on the input element, and add type instead?

**Comment by khmakoto** [comment]
Hi @eivindland I think you can already add `type='search'` today. Regarding modifying the `role` I don't think it's an outrageous demand and should be a relatively simple fix. @ecraig12345 what do you think about this?

**Comment by ecraig12345** [comment]
If this isn't working well with screen readers, we should fix the defaults rather than making them overridable. I'm also surprised that this hasn't been caught by previous accessibility test passes.

@eivindland What specific steps was your tester following, and what were the results (actual vs desired)? Does the same issue happen in our [SearchBox examples](https://developer.microsoft.com/en-us/fluentui#/controls/web/searchbox)?

**Comment by khmakoto** [comment]
@ecraig12345 well, he does say that this is with combinations other than Edge/Narrator, which makes me think that our defaults actually work in our primary supported scenario.

**Comment by eivindland** [comment]
Discussed this with our tester. There are two considertions here: 
1) The least serious is that it's recommened to use WAI-ARIA (role='searchbox') only where it's neccesary. Type='search' should be enough in this situation.
2) The most serious is that if role='searchbox' is present, the searchfields "disappear" when using the screen reader JAWS 18 in combination with IE11. It seems, however, to be working in Chrome and Firefox. And we get the same results on the Fluent example page. 

This is most likely a JAWS issue, but seeing these two issues together we think it would be best to change the attributes.
